service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/EventStreamConsumer.kt

```diff
@@ -63,6 +79,110 @@ class EventStreamConsumer(
+                if (toAddressBankUuid != null) {
+                    // TODO (steve) change to upsert
+                    transaction {
+                        CoinMovementRecord.insert(
+                            // TODO (steve) txhash is not unique in the general case so we have to fix this - but it works for now considering
+                            // we don't batch calls from this service at all
+                            txHash = event.txHash,
+                            fromAddress = event.memberId,
+                            fromAddressBankUuid = null,
+                            toAddress = event.withdrawAddress,
+                            toAddressBankUuid = toAddressBankUuid,
+                            blockHeight = event.height,
+                            blockTime = OffsetDateTime.parse(block.header.time),
+                            amount = event.amount,
+                            denom = event.withdrawDenom,
+                            type = MINT,
+                        )
+                    }
+                }
+            }
+
+        // SC Transfer events denote the "off ramp" for a bank user to redeem coin when the recipient is the bank address
+        batch.transfers(provenanceProperties.contractAddress)
+            .filter { it.sender.isNotEmpty() && it.recipient.isNotEmpty() }
+            .forEach { event ->
+                log.debug("SC Transfer - tx: ${event.txHash} sender: ${event.sender} recipient: ${event.recipient} amount: ${event.amount} denom: ${event.denom}")
+
+                // TODO (steve) implement caching
+                // TODO (steve) move to getAttributes by tag name
+                val fromAddressBankUuid = pbcService.getAttributes(event.sender).bankUuid()
+
+                // persist a record of this transaction if either the from or the to address has this bank's attribute
+                if (event.recipient == pbcService.managerAddress && fromAddressBankUuid != null) {
+                    // TODO (steve) change to upsert
+                    transaction {
+                        CoinMovementRecord.insert(
+                            txHash = event.txHash,
+                            fromAddress = event.sender,
+                            fromAddressBankUuid = fromAddressBankUuid,
+                            toAddress = event.recipient,
+                            toAddressBankUuid = null,
+                            blockHeight = event.height,
+                            blockTime = OffsetDateTime.parse(block.header.time),
+                            amount = event.amount,
+                            denom = event.denom,
+                            type = BURN,
+                        )
+                    }
+                }
+            }
+
+        // general coin transfers outside of the SC are tracked require the EventMarkerTransfer event
+        // TODO (steve) filter where denom is what we care about
+        batch.markerTransfers()
```

Comment:
does this filter out the burns above?

Reply:
Yes, totally different emitted events types.